apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/spi/defaultimpl/DefaultEmailService.java

```diff
@@ -49,6 +50,50 @@ public void send(Email email) {
       return;
     }
 
+    Boolean useSsl = portalConfig.emailConfigUseSsl();
+    if (useSsl) {
+      sendWithSsl(email);
+    } else {
+      sendWithoutSsl(email);
+    }
+  }
+
+  private void sendWithSsl(Email email) {
+    SMTPSSLTransport t = null;
+    try {
+      Properties prop = System.getProperties();
+      Session session = Session.getInstance(prop, null);
+
+      Message msg = new MimeMessage(session);
+      msg.setFrom(new InternetAddress(email.getSenderEmailAddress()));
+      msg.setRecipients(Message.RecipientType.TO, InternetAddress.parse(email.getRecipientsString(), false));
+      msg.setSubject(email.getSubject());
+      msg.setDataHandler(new DataHandler(new HTMLDataSource(email.getBody())));
+
+      String host = portalConfig.emailConfigHost();
+      String user = portalConfig.emailConfigUser();
+      String password = portalConfig.emailConfigPassword();
+
+      t = (SMTPSSLTransport) session.getTransport("smtps");
```

Comment:
It seems to me that the sole distinction between `sendWithSsl` and `sendWithoutSsl` lies in this line. If so, I propose adding an if statement here, rather than duplicating functionality across two methods.

Reply:
Thanks for your guidance.